### PR TITLE
feat(unlock-app): Add transfer page for walletless airdrop

### PR DESCRIPTION
**Pull request by searchableguy**
# Description

test

scr-

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@enzoferey/ethers-error-parser": "0.2.3",
```
Comment (julien51) [inline unlock-app/package.json:6]: Did you find this library to be good/useful?

File: unlock-app/src/components/helpers/SwitchNetwork.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  isOnRequiredNetwork: boolean
+  onNetworkChangeHandler: MouseEventHandler
+}
+
+interface Props {
+  requiredNetwork: number
+  children: (props: CProps) => JSX.Element
+}
+
+export const SwitchNetwork = ({ requiredNetwork, children }: Props) => {
```
Comment (julien51) [inline unlock-app/src/components/helpers/SwitchNetwork.tsx:14]: Interesting!

File: unlock-app/src/components/interface/transfer/index.tsx [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+  }
+
+  return (
+    <div className="grid gap-6 p-4 bg-white border rounded-lg">
+      <div className="space-y-1">
+        <h1 className="text-xl font-bold"> Request transfer code</h1>
+        <p className="text-gray-800">
+          You will be sent a transfer code by email. You can use this transfer
+          code to initiate the transfer. This code will expire in 15 minutes and
+          can only be used once on this page.
```
Comment (julien51) [inline unlock-app/src/components/interface/transfer/index.tsx:57]:
```suggestion
          You will receive a transfer code by email. You need to use this transfer
          code to transfer the membership token. This code will expire in 15 minutes and
          can only be used once, from this page.
```

File: unlock-app/src/components/interface/transfer/index.tsx [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+
+  return (
+    <div className="grid gap-6 p-4 bg-white border rounded-lg">
+      <div className="space-y-1">
+        <h1 className="text-xl font-bold">Confirm Transfer </h1>
+        <p className="text-gray-800">
+          Enter the transfer code you received by email to confirm the transfer.
+          This code will expire in 15 minutes and can only be used once on this
+          page. If you do not use the transfer code or it expires, you will need
+          to request a new one.
```
Comment (julien51) [inline unlock-app/src/components/interface/transfer/index.tsx:138]:
```suggestion
          Keep this window open and enter the transfer code you received by email to confirm the transfer.
          This code expires in 15 minutes and can only be used once from this
          page. If you do not use the transfer code or it expires, you can request a new one and start again.
```

**Review by julien51 — APPROVED** [review]

File: unlock-app/package.json (hunk 1 above)
Comment (searchableguy) [reply unlock-app/package.json:6]: It does parse some of the common errors better so somewhat useful but nothing with custom errors or wallet errors. Still searching for an overall good way to parse them into something more human readable without resorting to generic messages.

**Pull request merged**